addons/point_of_sale/static/src/app/debug/debug_widget.js

```diff
@@ -92,80 +90,89 @@ export class DebugWidget extends Component {
-                // this.pos.db.remove_all_unpaid_orders();
-                window.location = "/";
+                this.pos.data.resetUnsyncQueue();
+                const orders = this.pos.models["pos.order"].filter(
+                    (order) => order.finalized === paid
+                );
+
+                for (const order of orders) {
+                    this.pos.data.localDeleteCascade(order, true);
+                }
+
+                if (!this.pos.get_order()) {
+                    this.pos.add_new_order();
+                }
             },
         });
     }
-    _createBlob(contents) {
-        if (typeof contents !== "string") {
-            contents = JSON.stringify(contents, null, 2);
-        }
-        return new Blob([contents]);
-    }
-    // IMPROVEMENT: Duplicated codes for downloading paid and unpaid orders.
-    // The implementation can be better.
-    preparePaidOrders() {
-        try {
-            this.paidOrdersBlob = this._createBlob(this.pos.export_paid_orders());
-            this.state.isPaidOrdersReady = true;
-        } catch (error) {
-            console.warn(error);
-        }
-    }
-    get paidOrdersFilename() {
-        return `${_t("paid orders")} ${serializeDateTime(DateTime.now()).replace(
-            /:|\s/gi,
-            "-"
-        )}.json`;
-    }
-    get paidOrdersURL() {
-        var URL = window.URL || window.webkitURL;
-        return URL.createObjectURL(this.paidOrdersBlob);
-    }
-    // FIXME POSREF why is this two steps?
-    prepareUnpaidOrders() {
-        try {
-            this.unpaidOrdersBlob = this._createBlob(this.pos.export_unpaid_orders());
-            this.state.isUnpaidOrdersReady = true;
-        } catch (error) {
-            console.warn(error);
-        }
-    }
-    get unpaidOrdersFilename() {
-        return `${_t("unpaid orders")} ${serializeDateTime(DateTime.now()).replace(
-            /:|\s/gi,
-            "-"
-        )}.json`;
-    }
-    get unpaidOrdersURL() {
-        var URL = window.URL || window.webkitURL;
-        return URL.createObjectURL(this.unpaidOrdersBlob);
+    exportOrders(paid = false) {
+        const orders = this.pos.models["pos.order"]
+            .filter((order) => order.finalized === paid)
+            .map((o) => o.serialize({ orm: true }));
+
+        const blob = this._createBlob(orders);
+        const url = URL.createObjectURL(blob);
+        const a = document.createElement("a");
+        const fileName = `${paid ? "paid" : "unpaid"}_orders_${serializeDateTime(
+            DateTime.now()
+        ).replace(/:|\s/gi, "-")}.json`;
+
+        a.href = url;
+        a.download = fileName;
+        a.click();
     }
     async importOrders(event) {
         const file = event.target.files[0];
         if (file) {
-            const report = this.pos.import_orders(await file.text());
-            this.dialog.add(OrderImportPopup, { report });
+            const jsonData = JSON.parse(await file.text());
```

Comment:
will a file that was exported from pos in an older version ( ex 17 ) be able to be imported now in master?

Reply:
nop is over